pkg/controller/engine.go

```diff
@@ -196,11 +197,11 @@ func (e *Engine) Start(name string, o controller.Options, w ...Watch) error {
 
 	go func() {
 		<-e.mgr.Elected()
-		e.done(name, errors.Wrap(ca.Start(stop), errCrashCache))
+		e.done(name, errors.Wrap(ca.Start(context.Background()), errCrashCache))
```

Comment:
Yes, I added a new struct `stop` which holds the context as well as the cancel function. We now cancel the context on engine stop.

Reply:
@saschagrunert I am missing here where we are cancelling context on engine `Stop`, could you point it out to me? It looks like we still close the stop channel, which is an empty struct.

Reply:
Ah this has been reverted after my rebase on top of 0.8.0. We have to provide a context, which will be then passed around. The runtime docs says that it will block until the context gets cancelled, so I'll now use the cancelfunc of the context instead of a close channel. WDYT?

Reply:
Looks pretty nice to me :) Let me do a little testing just to validate my assumptions here :+1:

Reply:
Awesome, thank you! :pray:

Reply:
@hasheddan I've also opened https://github.com/crossplane/crossplane/pull/2082 which might facilitate helping.

Reply:
Awesome, thanks so much @khos2ow!